cmd/fry.go

```diff
@@ -19,69 +19,132 @@ import (
 	"github.com/sonatype-nexus-community/hashbrowns/types"
 
 	"github.com/sonatype-nexus-community/nancy/cyclonedx"
 	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
 )
 
+var log *logrus.Logger
+
 // fryCmd represents the fry command
 var fryCmd = &cobra.Command{
 	Use:   "fry",
 	Short: "Submit list of sha1s to Nexus IQ Server",
 	Long: `Provided a path to a file with sha1's and locations, this command will submit them to Nexus IQ Server.
 
 This can be used to audit generic environments for matches to known hashes that do not meet your org's policy.`,
+	SilenceErrors: true,
 	RunE: func(cmd *cobra.Command, args []string) (err error) {
+		defer func() {
+			if r := recover(); r != nil {
+				var ok bool
+				err, ok = r.(error)
+				if !ok {
+					err = fmt.Errorf("pkg: %v", r)
+				}
+
+				logger.PrintErrorAndLogLocation(err)
+			}
+		}()
+
+		fflags := cmd.Flags()
+
+		checkRequiredFlags(fflags)
+
+		log = logger.GetLogger("", config.LogLevel)
+
+		log.Info("Running Fry Command")
+
 		var exitCode int
 		if exitCode, err = doParseSha1List(&config); err != nil {
 			return
-		} else {
-			// TODO use something like ErrorExit custom error to pass up exit code, instead of calling os.Exit() here
-			os.Exit(exitCode)
 		}
-		return
+
+		if exitCode == 0 {
+			return
+		}
+
+		return fmt.Errorf("Non zero exit code: %d", exitCode)
 	},
 }
 
 func init() {
 	rootCmd.AddCommand(fryCmd)
 
-	fryCmd.PersistentFlags().StringVar(&config.Path, "path", "", "Path to file with sha1s")
-	fryCmd.PersistentFlags().StringVar(&config.User, "user", "admin", "Specify Nexus IQ username for request")
-	fryCmd.PersistentFlags().StringVar(&config.Token, "token", "admin123", "Specify Nexus IQ token/password for request")
-	fryCmd.PersistentFlags().StringVar(&config.Server, "server-url", "http://localhost:8070", "Specify Nexus IQ Server URL")
-	fryCmd.PersistentFlags().StringVar(&config.Application, "application", "", "Specify application ID for request")
-	fryCmd.PersistentFlags().StringVar(&config.Stage, "stage", "develop", "Specify stage for application")
-	fryCmd.PersistentFlags().IntVar(&config.MaxRetries, "max-retries", 300, "Specify maximum number of tries to poll Nexus IQ Server")
+	pf := fryCmd.PersistentFlags()
+
+	pf.StringVar(&config.Path, "path", "", "Path to file with sha1s (required)")
+	pf.StringVar(&config.User, "user", "admin", "Specify Nexus IQ username for request")
+	pf.StringVar(&config.Token, "token", "admin123", "Specify Nexus IQ token/password for request")
+	pf.StringVar(&config.Server, "server-url", "http://localhost:8070", "Specify Nexus IQ Server URL")
+	pf.StringVar(&config.Application, "application", "", "Specify application ID for request (required)")
+	pf.StringVar(&config.Stage, "stage", "develop", "Specify stage for application")
+	pf.IntVar(&config.MaxRetries, "max-retries", 300, "Specify maximum number of tries to poll Nexus IQ Server")
+}
+
+func checkRequiredFlags(flags *pflag.FlagSet) {
+	if !flags.Changed("path") {
```

Comment:
Interesting how "required" works with Cobra - not hugely intuitive, but I think I get it now.

Reply:
Yeah, it doesn't really output what you are missing, just shows you usage. Not super helpful. I did this as a result of that.